src/transformers/training_args.py

```diff
@@ -663,8 +663,19 @@ def __post_init__(self):
         self.lr_scheduler_type = SchedulerType(self.lr_scheduler_type)
         if self.do_eval is False and self.evaluation_strategy != IntervalStrategy.NO:
             self.do_eval = True
-        if self.eval_steps is None:
-            self.eval_steps = self.logging_steps
+
+        # eval_steps has to be defined and non-zero, fallbacks to logging_steps if the latter is non-zero
+        if self.do_eval and (self.eval_steps is None or self.eval_steps == 0):
```

Comment:
`do_eval` is not the right check to add here: `do_eval` means do an evaluation after training in the example scripts, it should not control any parameter of the `Trainer` (and is documented as not impacting the `Trainer`). The right check is `self.evaluation_strategy == IntervalStrategy.STEPS`.